src/lib.rs

```diff
@@ -145,6 +161,26 @@ impl Sdl2Window {
         Ok(window)
     }
 
+    /// Initialize the joystick subsystem. Required before joystick input
+    /// events will be returned. Returns the number available or error.
+    pub fn init_joysticks(&mut self) -> Result<u32, String> {
```

Comment:
yes. It's stored in the state, so we can open up an accessor to it. I was deliberately conservative on the API though until I understand more about how joysticks work (like, does it even make sense to expose this like this in the face of re-connecting devices?)